### PR TITLE
CLDR-16499 Prepare to Require a CLA for contributions

### PR DESCRIPTION
**Pull request by srl295**
CLDR-16499

For v46, we do NOT require a CLA, but leave the code in place for the future

- property `REQUIRE_CLA` off by default, if on it will enable the flow
- no popups or menu items visible to users
- enables loading .md files to the front end and as assets
- improve cldrClient.getClient() to cache, speedup repeated calls
- add notice for synchronizing webpack configuration
- APIs for sign/revoke/read CLA
- web flow for signing CLA and handling various cases

- [ ] This PR completes the ticket.

ALLOW_MANY_COMMITS=true

### CODE REVIEW

**Comment by srl295** [comment]
@btangmu i think this could get any feedback from you if you want to provide it, knowing we still need to review the text and the overall process. But let me know if you see any structural issues.

File: tools/cldr-apps/js/src/views/MainMenu.vue
Comment (btangmu) [inline tools/cldr-apps/js/src/views/MainMenu.vue:11]: what's this? better not to add commented-out code?

File: tools/cldr-apps/js/src/views/SignCla.vue
Comment (btangmu) [inline tools/cldr-apps/js/src/views/SignCla.vue:98]: these imports should be alphabetized, better than random order

File: tools/cldr-apps/js/src/views/SignCla.vue
Comment (btangmu) [inline tools/cldr-apps/js/src/views/SignCla.vue:121]: for separation of concerns, my preference is that Vue components shouldn't include client/server code

I realize you may not agree with that, and I don't know if I could persuade you, so I won't insist on it :-)

**Review by btangmu — CHANGES_REQUESTED** [review]
remove the commented-out code change for MainMenu.vue?

I made a couple other nit-picky comments (which I don't insist on), otherwise, it all looks great

File: tools/cldr-apps/js/src/views/SignCla.vue
Comment (srl295) [reply tools/cldr-apps/js/src/views/SignCla.vue:121]: This is a draft PR for the purpose of a UX / legal review. I appreciate the comments and will do them to refactor before merge. I do plan to merge with everything switched off so we can un mothball it later.

Comment (srl295) [reply tools/cldr-apps/js/src/views/SignCla.vue:121]: done

**Comment by srl295** [comment]
> remove the commented-out code change for MainMenu.vue?
> 
> I made a couple other nit-picky comments (which I don't insist on), otherwise, it all looks great

the menu item was removed because it's not relevant while CLAs are not required. I've changed it to being driven by a constant.

**Comment by srl295** [comment]
@btangmu ready to review for merging and mothballing.

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/UserAPI.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/UserAPI.java:49]: TODO should have ticket link; and it's not clear what needs to be done, if anything; false makes sense for a logged-out user

File: tools/cldr-apps/js/src/esm/cldrCla.mjs
Comment (btangmu) [inline tools/cldr-apps/js/src/esm/cldrCla.mjs:12]: on the back end, it looks like "signed" is a Date rather than boolean?

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/ClaSignature.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/ClaSignature.java:22]: Date not boolean, but front-end comments say it's boolean

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/UserRegistry.java:616]: this constructor presumably sets signed = true or signed = Date... OK, I've confirmed this suspicion

File: tools/cldr-apps/src/main/java/org/unicode/cldr/web/ClaSignature.java
Comment (btangmu) [inline tools/cldr-apps/src/main/java/org/unicode/cldr/web/ClaSignature.java:69]: OK, here's the constructor used for DO_NOT_REQURE_CLA, setting signed = Date, which is treated as true

File: tools/cldr-apps/js/src/views/MainHeader.vue
Comment (btangmu) [inline tools/cldr-apps/js/src/views/MainHeader.vue:166]: I'm guessing that if DO_NOT_REQURE_CLA is true on the back end, then user.claSigned will get true (or truthy Date object?), even though that's sort of a fiction... OK, I've confirmed that

**Review by btangmu — DISMISSED** [review]
Looks great, except I suggest renaming "signed" to "dateSigned", since it's a Date, not a boolean (unless I still misunderstand), and fixing the front-end comment saying it's boolean

I've left a few comments/questions I had while studying the PR, even though I was eventually able to answer them for myself. I think it would help to add some clarifying comments for maintainability. On the front end, "user.claSigned" really means the user has signed OR doesn't need to because DO_NOT_REQURE_CLA

File: tools/cldr-apps/js/src/esm/cldrCla.mjs
Comment (srl295) [reply tools/cldr-apps/js/src/esm/cldrCla.mjs:12]: good catch, that should be the date signed.

**Comment by srl295** [comment]
@btangmu @macchiati this is ready to go in. it is disabled by default, but gets all the mechanism for requiring a CLA in place (just pending wording changes)

File: tools/cldr-apps/js/src/views/MainHeader.vue
Comment (srl295) [reply tools/cldr-apps/js/src/views/MainHeader.vue:186]: yes

**Review by macchiati — APPROVED** [review]
Sgtm. Tom should review details

**Comment by srl295** [comment]
@macchiati Tom had previously approved, just had merge conflicts.

File: tools/cldr-apps/js/src/esm/cldrCla.mjs
Comment (btangmu) [reply tools/cldr-apps/js/src/esm/cldrCla.mjs:12]: should that still be changed to a date, then?

**Pull request merged**
